### PR TITLE
Added support for extracting RGB data to python.

### PR DESCRIPTION
**Pull request by m-kuoch**
Now properly extracts RBG data to the python dictionary.

### CODE REVIEW

File: pylcm/__init__.py [hunk 1]
```diff
@@ -17,6 +17,12 @@
 import scipy.io.matlab.mio
 from lcm import EventLog
 from .scan_for_lcmtypes import make_lcmtype_dictionary
+import robotlocomotion as rl
+import inspect
+import imageio
+import numpy as np
+from io import BytesIO
```
Comment (duanyutong) [inline pylcm/__init__.py:24]: removed unused imports

File: pylcm/__init__.py [hunk 2]
```diff
@@ -64,35 +70,48 @@ def msg_to_dict (data, e_channel, msg, statusMsg, verbose=False, lcm_timestamp=-
...
         for i in range(len(constants)):
-            myValue = None
-            myValue = eval('msg.' + constants[i])
-            data[e_channel][constants[i][:31]] = myValue
+            my_value = None
+            my_value = eval('msg.' + constants[i])
+            data[e_channel][constants[i][:31]] = my_value
 
     # Get lcm fields and constants
     fields = msg_getfields (msg)
```
Comment (duanyutong) [inline pylcm/__init__.py:78]: remove extra space

File: pylcm/__init__.py [hunk 3]
```diff
@@ -64,35 +70,48 @@ def msg_to_dict (data, e_channel, msg, statusMsg, verbose=False, lcm_timestamp=-
...
 
     # Iterate each field of the LCM message
     for i in range(len(fields)):
```
Comment (duanyutong) [inline pylcm/__init__.py:81]: for field in fields since `i` isn't used

File: pylcm/__init__.py [hunk 4]
```diff
@@ -64,35 +70,48 @@ def msg_to_dict (data, e_channel, msg, statusMsg, verbose=False, lcm_timestamp=-
         # Iterate each constant of the LCM message
         constants = msg_getconstants (msg)
         for i in range(len(constants)):
```
Comment (duanyutong) [inline pylcm/__init__.py:72]: for const in constants

File: pylcm/__init__.py [hunk 5]
```diff
@@ -55,7 +61,7 @@ def msg_getconstants (lcm_msg):
     return constantslist
 
 
-def msg_to_dict (data, e_channel, msg, statusMsg, verbose=False, lcm_timestamp=-1):
```
Comment (duanyutong) [inline pylcm/__init__.py:58]: could you set up black, flake8 and pylint as pre-commit hooks for this repo as done in `dexai-python`? there's lot of styling issues

File: pylcm/__init__.py [hunk 6]
```diff
@@ -64,35 +70,48 @@ def msg_to_dict (data, e_channel, msg, statusMsg, verbose=False, lcm_timestamp=-
...
-            msg_to_dict (data[e_channel], fields[i][:31], submsg, statusMsg, verbose)
-
+            msg_to_dict (data[e_channel], fields[i][:31], submsg, status_msg, verbose)
+ 
+        # Handles getting RBGD data from 'images' field
+        elif (fields[i] == "images" and 
+              isinstance(my_value, list) and 
+              isinstance(my_value[0], image_t)):
+            # Read image_t.data to numpy arrays
+            color_image = np.array(imageio.imread(my_value[0].data))
```
Comment (duanyutong) [inline pylcm/__init__.py:104]: rename variable to rgb_img, why only read the first item? what are the rest? are they handled elsewhere or not rgb data and discarded?

File: pylcm/__init__.py [hunk 7]
```diff
@@ -64,35 +70,48 @@ def msg_to_dict (data, e_channel, msg, statusMsg, verbose=False, lcm_timestamp=-
...
+            #depth_image = np.array(imageio.imread(my_value[1].data))
```
Comment (duanyutong) [inline pylcm/__init__.py:105]: how long is `my_value`? is `my_value[1]` a depth image? can you read it?

File: pylcm/__init__.py [hunk 8]
```diff
@@ -64,35 +70,48 @@ def msg_to_dict (data, e_channel, msg, statusMsg, verbose=False, lcm_timestamp=-
...
+            try:
+                data[e_channel]['RGB'].append(color_image)
+                #data[e_channel]['Depth'].append(depth_image)
+            except KeyError as AttributeError:
+                data[e_channel]['RGB'] = [(color_image)]
```
Comment (duanyutong) [inline pylcm/__init__.py:110]: why `[(color_image)]` rather than `[color_image]`

File: pylcm/__init__.py [hunk 9]
```diff
@@ -64,35 +70,48 @@ def msg_to_dict (data, e_channel, msg, statusMsg, verbose=False, lcm_timestamp=-
...
+        elif (fields[i] == "images" and 
+              isinstance(my_value, list) and 
+              isinstance(my_value[0], image_t)):
+            # Read image_t.data to numpy arrays
+            color_image = np.array(imageio.imread(my_value[0].data))
+            #depth_image = np.array(imageio.imread(my_value[1].data))
+            try:
+                data[e_channel]['RGB'].append(color_image)
+                #data[e_channel]['Depth'].append(depth_image)
+            except KeyError as AttributeError:
```
Comment (duanyutong) [inline pylcm/__init__.py:109]: you don't need ` as AttributeError`

File: pylcm/__init__.py [hunk 10]
```diff
@@ -128,6 +146,7 @@ def parse_lcm(fname, opts=None):
     checkIgnore = False
     channelsToProcess = ".*"
     separator = ' '
+    returnDict = False
```
Comment (duanyutong) [inline pylcm/__init__.py:149]: change to snake case

File: pylcm/__init__.py (hunk 7 above)
Comment (m-kuoch) [reply pylcm/__init__.py:105]: `my_value` has len=2, one rgb and one depth image.  `my_value[1]` is a depth image, but reading it the same way gives an error.  Don't now how to properly read it.

File: pylcm/__init__.py (hunk 8 above)
Comment (m-kuoch) [reply pylcm/__init__.py:110]: Not sure, but I was following what was originally in the code.

File: pylcm/__init__.py (hunk 6 above)
Comment (m-kuoch) [reply pylcm/__init__.py:104]: The first item represents the RGB data.  The second item represents the depth image.

File: pylcm/__init__.py (hunk 5 above)
Comment (m-kuoch) [reply pylcm/__init__.py:58]: I'll do this now.

**Comment by duanyutong** [comment]
Posted depth image solution on slack, marking conversations regarding depth as resolved and to be added in next PR

**Pull request merged**